### PR TITLE
Video playback support via network

### PR DESCRIPTION
**Pull request by MayuriKhinvasara**
Support playback of video through network. Modify the MediaMetadataRetriever data sources to support remote urls. Move this to an IO Thread.

### CODE REVIEW

**Comment by calren** [comment]
I'm getting a crash when I try to set the media uri: https://storage.googleapis.com/exoplayer-test-media-1/mp4/android-screens-10s.mp4

In `Contact.kt` I set the dog to reply with a mediaUri like this:
```
    "dog" to { _ ->
        buildReply {
            this.text = "Check out the Top 3 Modern Android Development Announcements from I/O '23!"
            mediaUri = "https://storage.googleapis.com/exoplayer-test-media-1/mp4/android-screens-10s.mp4"
            mediaMimeType = "video/mp4"
        }
    },
```

but this leads to a crash 

```
  java.lang.RuntimeException: setDataSource failed: status = 0xFFFFFFEA
                                                                                                    	at android.media.MediaMetadataRetriever._setDataSource(Native Method)
                                                                                                    	at android.media.MediaMetadataRetriever.setDataSource(MediaMetadataRetriever.java:281)
```

Should I use another media uri to test remote videos?

**Comment by calren** [comment]
ah, i think the crash is due to local videos, not remote videos.

i added some print statements. when i record a video, the uri is `/storage/emulated/0/Android/data/com.google.android.samples.socialite/cache/Socialite-edited-recording-2024-08-05-12-09-45-454.mp4` , which doesn't start with `content://`

**Comment by MayuriKhinvasara** [comment]
Fixed the crash. Switched the logic to tread remote urls via https.

File: app/src/main/java/com/google/android/samples/socialite/ui/home/timeline/Timeline.kt [hunk 1]
```diff
@@ -238,15 +242,22 @@ fun MetadataOverlay(modifier: Modifier, mediaItem: TimelineMediaItem) {
...
-                Uri.parse(mediaItem.uri),
-            )
+            val context = LocalContext.current.applicationContext
 
-            val duration =
-                mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)
-                    ?.toLong()
-            duration?.let {
+            val duration: State<Long?> = produceState<Long?>(initialValue = null) {
+                withContext(Dispatchers.IO) {
```
Comment (calren) [inline app/src/main/java/com/google/android/samples/socialite/ui/home/timeline/Timeline.kt:248]: are we doing this on the IO thread to reduce lag time when loading the metadata from a remote url?

if so, can we add a quick comment here explaining why we're doing this on the IO thread

**Comment by calren** [comment]
lgtm, one suggestion about adding a comment for why we moved getting the video duration to the IO thread, as I think it's not clear for a developer just reading the code

**Review by calren — APPROVED** [review]

**Pull request merged**